modules/fluid_properties/include/utils/NewtonInversion.h

```diff
@@ -0,0 +1,64 @@
+// * This file is part of the MOOSE framework
+// * https://www.mooseframework.org
+// *
+// * All rights reserved, see COPYRIGHT for full restrictions
+// * https://github.com/idaholab/moose/blob/master/COPYRIGHT
+// *
+// * Licensed under LGPL 2.1, please see LICENSE for details
+// * https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "Moose.h"
+
+// C++ includes
+#include "libmesh/utility.h"
+#include <functional>
+
+// This file performs a Newton solve in 1D and 2D. In 1D, solve fails to converge
+// and error outputs if it takes more than 100 iterations. In 2D, if more than 100
+// iterations occur, save results references and set convergence to false. Applications
+// using 2D Newton Solve will need to write routines to take care of values where this
+// occurs. See TabulatedBicubicFluidProperties for example (routine checkOutofBounds).
```

Comment:
This file should get a corresponding doc file. You can refer to it on your other doc pages.